haystack/preview/components/classifiers/file_classifier.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import mimetypes
+import re
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Union, Optional
+
+from haystack.preview import component
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class FileTypeClassifier:
+    """
+    A component that classifies files based on their MIME types.
+
+    The FileTypeClassifier takes a list of file paths and groups them by their MIME types.
+    The list of MIME types to consider is provided during the initialization of the component.
+
+    This component is particularly useful when working with a large number of files, and you
+    want to categorize them based on their MIME types.
+    """
+
+    def __init__(self, mime_types: List[str]):
+        """
+        Initialize the FileTypeClassifier.
+
+        :param mime_types: A list of file mime types to consider when classifying
+        files (e.g. ["text/plain", "audio/x-wav", "image/jpeg"]).
+        """
+        if not mime_types:
+            raise ValueError("The list of mime types cannot be empty.")
+
+        all_known_mime_types = all(self.is_known_mime_type(mime_type) for mime_type in mime_types)
+        if not all_known_mime_types:
+            raise ValueError(f"The list of mime types contains unknown mime types: {mime_types}")
+
+        # convert the mime types to the underscore format (e.g. text_plain)
+        # otherwise we'll have issues with the dataclass field name convention
+        # in the output dataclass
+        mime_types = [self.to_underscore_format(mime_type) for mime_type in mime_types]
```

Comment:
We're not creating dataclasses anymore with the new components API: it's all dictionaries now. So probably this conversion is not needed anymore!